### PR TITLE
DAO app

### PR DESCRIPTION
**Pull request by Macket**
Changes:
- Added dao app to curve.fi

### CODE REVIEW

**Comment by Macket** [comment]
image
Infinite loading on gauges page

This is the reason

`Access to fetch at 'https://prices.curve.fi/v1/dao/gauges/overview' from origin 'https://staging-dao.curve.fi' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by bout3fiddy** [comment]
telegram-cloud-photo-size-4-5978664823757586793-y

active page should arrange proposals by end date so more urgent proposals are on the top, and not on the bottom

**Comment by mimaklas** [comment]
image
It would be good to show something similar to the old UI "enacted",(for a passed and enacted/executed vote) without requiring users to go into the detailed view.
image

File: apps/dao/lingui.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  locales: ['en', 'zh-Hans', 'zh-Hant', 'pseudo'],
+  pseudoLocale: 'pseudo',
+  sourceLocale: 'en',
+  fallbackLocales: {
+    default: 'en',
+  },
+  catalogs: [
+    {
+      path: 'src/locales/{locale}/messages',
+      include: ['src/components', 'src/hooks', 'src/layout', 'src/pages'],
```
Comment (DanielSchiavini) [inline apps/dao/lingui.config.js:11]: Don't forget the ui kit here

File: apps/dao/public/fonts/commodore/READ_ME.txt
Comment (DanielSchiavini) [inline apps/dao/public/fonts/commodore/READ_ME.txt:1]: Commodore has been deleted

File: apps/dao/src/components/AlertFormError.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import type { AlertBoxProps } from '@/ui/AlertBox/types'
+
+import { t } from '@lingui/macro'
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+
+import AlertBox from '@/ui/AlertBox'
+
+const ALERT_FORM_ERROR_KEYS = {
+  'error-user-rejected-action': 'error-user-rejected-action',
```
Comment (DanielSchiavini) [inline apps/dao/src/components/AlertFormError.tsx:10]: Looks like an enum imo

File: apps/dao/src/components/Charts/BarChartComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import styled from 'styled-components'
+import { BarChart, Bar, Cell, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, TooltipProps } from 'recharts'
+
+type Props<T> = {
+  data: T[]
+  dataKey: keyof T & string
+  CustomTooltip: React.ComponentType<TooltipProps<any, any>>
+}
+
+const COLORS = [
```
Comment (DanielSchiavini) [inline apps/dao/src/components/Charts/BarChartComponent.tsx:10]: Should these come from the palette or css variables?

File: apps/dao/public/fonts/commodore/READ_ME.txt
Comment (DanielSchiavini) [reply apps/dao/public/fonts/commodore/READ_ME.txt:1]: It isn't merged yet. We can do that separately

File: apps/dao/lingui.config.js (hunk 1 above)
Comment (DanielSchiavini) [reply apps/dao/lingui.config.js:11]: I just merged the menu PR, sorry

File: apps/dao/src/components/ConnectWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const loading = isLoading(connectState)
+
+  return (
+    <ConnectWalletPrompt
+      connectWallet={updateConnectWalletStateKeys}
+      description={description}
+      connectText={connectText}
+      loadingText={loadingText}
+      isLoading={loading}
+      theme={theme === 'dark' ? 'dark' : 'light'}
```
Comment (DanielSchiavini) [inline apps/dao/src/components/ConnectWallet.tsx:26]: What happened to chad?

File: apps/dao/src/components/Page404/index.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import styled from 'styled-components'
+
+import { breakpoints } from '@/ui/utils/responsive'
+import Box from '@/ui/Box'
+
+export function ErrorVideo() {
+  const videoRef = useRef<HTMLVideoElement>(null)
+
+  useEffect(() => {
+    if (videoRef.current) {
```
Comment (DanielSchiavini) [inline apps/dao/src/components/Page404/index.tsx:11]: Either the if should be removed or we need it in the dependencies..?

File: apps/dao/src/components/PageAnalytics/CrvStats/index.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  const veCrvFeesLoading = veCrvFees.fetchStatus === 'LOADING'
+
+  useEffect(() => {
+    if (provider && veCrvData.totalCrv === 0 && veCrvData.fetchStatus !== 'ERROR') {
+      getVeCrvData(provider)
+    }
+  }, [veCrvData.totalCrv, veCrvData.fetchStatus, getVeCrvData, provider])
+
+  const veCrvApr =
+    veCrvLoading || veCrvFeesLoading ? 0 : (((veCrvFees.fees[1].fees_usd / veCrvData.totalVeCrv) * 52) / 0.3) * 100
```
Comment (DanielSchiavini) [inline apps/dao/src/components/PageAnalytics/CrvStats/index.tsx:27]: Doesn't look like the kind of calculation that should be in a component 

How about a hook or helper fn?

File: apps/dao/src/components/ConnectWallet.tsx (hunk 4 above)
Comment (OnlyJousting) [reply apps/dao/src/components/ConnectWallet.tsx:26]: dark and light refers to the two different variants of the background image, light and chad shares the same

File: apps/dao/src/components/Page404/index.tsx (hunk 5 above)
Comment (OnlyJousting) [reply apps/dao/src/components/Page404/index.tsx:11]: this is some legacy code, I removed it

**Review by OnlyJousting — APPROVED** [review]

**Review by bout3fiddy — APPROVED** [review]

**Pull request merged**